### ISSUE TITLE
Consider changing the position of the `OPTIONS` clause and renaming it

### ISSUE THREAD

**Issue by vmarcos** [MaterializeInc/materialize#21531]
### Feature request

[It has been remarked](https://github.com/MaterializeInc/materialize/pull/21500#discussion_r1312837094) that the position in which we apply hints presently, in the `OPTIONS` clause of a `SELECT` right after the `GROUP BY` clause, is potentially confusing. We should consider whether we can move it to the end of the `SELECT` query block and potentially rename, e.g., as `QUERY HINTS`. One important issue is that when adding this feature, there should be a strategy for backwards compatibility, e.g., keeping the previous syntax also redundantly supported and producing an error when the clauses are used in the two positions simultaneously.

**Comment by vmarcos** [comment]
[Some relevant discussion](https://github.com/MaterializeInc/materialize/pull/21500#discussion_r1312914884).

**Comment by antiguru** [comment]
Related: #20803 which could help us avoid query hints.

**Comment by vmarcos** [comment]
I agree that eventually cardinality estimations (if refined further from #19649) would be helpful to reduce the use of query hints. Exploiting #20803 might be a bit tricky. The issue is that if we perform a hierarchical reduction, we need to arrange the input to the hierarchical reduction after we apply the `KeyValPlan`, negating the possibility to reuse an index for the first level of the hierarchical aggregation. This happens now even if the `KeyValPlan` is an identity mapping, which could potentially be improved.
